package/batocera/core/batocera-configgen/scripts/powermode_launch_hooks.sh

```diff
@@ -1,145 +1,97 @@
-
-  local AVAILABLE_GOVERNORS=$(cat "$AVAILABLE_GOVERNORS_PATH")
-
-  if echo "$AVAILABLE_GOVERNORS" | grep -q "\b$GOVERNOR_TO_CHECK\b"; then
-    return 0
-  else
-    return 1
-  fi
-}
-
-# Set governor
-set_governor() {
-    local GOVERNOR_NAME=$1
-
-    # Apply the governor to all policies
-    for policy in /sys/devices/system/cpu/cpufreq/policy*; do
-        if [ -e "$policy/scaling_governor" ]; then
-            local CURRENT_GOVERNOR=$(cat "$policy/scaling_governor")
-            if [ "$CURRENT_GOVERNOR" != "$GOVERNOR_NAME" ]; then
-                echo $GOVERNOR_NAME > "$policy/scaling_governor" 2>/dev/null
-            fi
-        fi
-    done
-}
+is_power_connected() {
+    # Detect battery directory
+    BATTERY_DIR=$(ls -d /sys/class/power_supply/*{BAT,bat}* 2>/dev/null | head -1)
 
-# Check if Energy Performance Preferences are available
-epp_available() {
-    if [ -e /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_available_preferences ]; then
+    if [ -z "${BATTERY_DIR}" ]; then
+        # If no battery directory, assume power supply is connected
         return 0
-    else
-        return 1
     fi
-}
 
-# Set EPP
-set_epp() {
-    local PREFERENCE=$1
-    local AVAILABLE_PREFERENCES
-    AVAILABLE_PREFERENCES=$(cat /sys/devices/system/cpu/cpu0/cpufreq/energy_performance_available_preferences)
+    # Check the battery status
+    BATTERY_STATUS=$(cat "${BATTERY_DIR}/status" 2>/dev/null)
 
-    # Check if the specified preference is available
-    if ! echo "$AVAILABLE_PREFERENCES" | grep -q "\b$PREFERENCE\b"; then
+    if [[ "${BATTERY_STATUS}" == "Discharging" ]]; then
+        # Battery is discharging
         return 1
+    else
+        # Battery is not discharging, assume power supply is connected
+        return 0
     fi
-
-    for cpu in /sys/devices/system/cpu/cpu*/cpufreq/energy_performance_preference; do
-        if [ -e "$cpu" ]; then
-            echo "$PREFERENCE" > "$cpu" 2>/dev/null
-        fi
-    done
 }
 
-is_power_connected() {
-    # Detect battery
-    BATTERY_DIR=$(ls -d /sys/class/power_supply/*{BAT,bat}* 2>/dev/null | head -1)
-    BATT=$(cat ${BATTERY_DIR}/uevent 2>/dev/null | grep -E "^POWER_SUPPLY_CAPACITY=" | sed -e s+'^POWER_SUPPLY_CAPACITY='++ | sort -rn | head -1)
-    PLUGGED=$(cat /sys/class/power_supply/*/online 2>/dev/null | grep -E "^1")
-    if [ -n "${PLUGGED}" ]; then
-        echo 1  # Power supply is connected
-    elif [ -z "${BATT}" ]; then
-        echo 1  # No battery detected, assume power supply is connected
+handle_game_stop() {
+    # Check if power connected
+    if is_power_connected; then
+        POWER_MODE="$(/usr/bin/batocera-settings-get-master global.powermode)"
     else
-        echo 0  # Power supply is not connected
+        POWER_MODE="$(/usr/bin/batocera-settings-get-master global.batterymode)"
+    fi
+
+    # Apply global power mode or fall back to default
+    if [ -n "${POWER_MODE}" ]; then
```

Comment:
Why not this?

```sh
/usr/bin/batocera-power-mode "${POWER_MODE:-default}"
```

Reply:
I'll update the script to include it. It simplifies it a bit.